### PR TITLE
ref(gsh): Move GSH initialization to gsh/index w/ useEffect

### PR DESCRIPTION
**Pull request by evanpurkhiser**
The InitalizeGlobalSelectionHeader component was only here to do `componentDidMount` and `componentDidUpdate` things, which we can just use `useEffect` for.

### CODE REVIEW

File: static/app/components/organizations/globalSelectionHeader/index.tsx [hunk 1]
```diff
@@ -53,41 +71,92 @@ function GlobalSelectionHeaderContainer({
...
 
-  const enforceSingleProject = !organization.features.includes('global-views');
+  // Initializes GlobalSelectionHeader
+  //
+  // Calls an actionCreator to load project/environment from local storage if possible,
+  // otherwise populate with defaults.
+  //
+  // This should only happen when the header is mounted
+  // e.g. when changing views or organizations.
+  useEffect(() => {
```
Comment (evanpurkhiser) [inline static/app/components/organizations/globalSelectionHeader/index.tsx:81]: if we really wanna get fancy we can move this into it's own hook

Comment (markstory) [reply static/app/components/organizations/globalSelectionHeader/index.tsx:81]: That seems elaborate given we have no other use cases for the custom hook.

**Review by markstory — APPROVED** [review]
Looks good to me.

File: static/app/components/organizations/globalSelectionHeader/index.tsx (hunk 1 above)
Comment (evanpurkhiser) [reply static/app/components/organizations/globalSelectionHeader/index.tsx:81]: yeah agreed there.

Was suggesting this more as a way to encapsulate this logic in a different file

**Pull request merged**